File: java-example/src/test/java/ru/stqa/training/selenium/CustomerListPage.java
package ru.stqa.training.selenium;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

import java.util.Set;

import static java.util.stream.Collectors.toSet;

public class CustomerListPage extends Page {

    public CustomerListPage(WebDriver driver) {
        super(driver);
    }

    public CustomerListPage open() {
        driver.get("http://localhost/litecart/admin/?app=customers&doc=customers");
        return this;
    }


    public Set<String> getCustomerIds() {
        return driver.findElements(By.cssSelector("table.dataTable tr.row")).stream()
                .map(e -> e.findElements(By.tagName("td")).get(2).getText())
                .collect(toSet());
    }
}

File: js-example/pages/customer_list_page.js
var webdriver = require('selenium-webdriver');

var By = webdriver.By,
    until = webdriver.until;

class CustomerListPage {

    constructor(driver) {
        this.driver = driver;
    }

    open() {
        this.driver.get("http://localhost/litecart/admin/?app=customers&doc=customers");
        return this;
    }

    customerRows() {
        return this.driver.findElements(By.css("table.dataTable tr.row"));
    }

    getCustomerIds() {
        var ids = [];
        return this.customerRows().then(function(rows) {
            for (var i=0; i < rows.length; i++) {
                rows[i].findElements(By.css("td")).then(function(cells) {
                    cells[2].getText().then(function(txt) {
                        ids.push(txt);
                    });
                });
            }
        });
        return ids;
    }

}

exports.CustomerListPage = CustomerListPage;

